File: is-palindrome/Palindrome.swift
import Foundation 

class Palindrome {
    func isPalindrome(str: String) -> Bool {
        var array: [Character] = Array(str)
        var left = 0
        var right = str.count - 1
        while(left < right) {
            if(array[left].lowercased() != array[right].lowercased()) { return false }
            left += 1; right -= 1;
        }
        return true
    }

    func isPalindrome2(str: String) -> Bool {
        var array: [Character] = Array(str)
        var left = 0
        var right = str.count - 1
        while(left < right) {
            while left < right , !isLetterOrDigit(array[left]) { left += 1 }
            while left < right , !isLetterOrDigit(array[right]) { right -= 1 }
            if(array[left].lowercased() != array[right].lowercased()) { return false }
            left += 1; right -= 1;
        }
        return true
    }

    func isLetterOrDigit(_ character: Character) -> Bool {
        return character.isLetter || character.isNumber
    }
}

File: is-palindrome/Palindrome.cs
using System;
using System.Collections.Generic;
public class Palindrome {
    public static bool IsPalindrome1(string str) 
    {
        int left = 0;
        int right = str.Length - 1;
        while (left < right)
        {
            if (Char.ToLower(str[left]) != Char.ToLower(str[right])) { return false; }
            left++; right--;
        }
        return true;
    }

    public static bool IsPalindrome2(string str) 
    {
        int left = 0;
        int right = str.Length - 1;
        while(left < right) {
            while(left < right && !Char.IsLetterOrDigit(str[left])) { left++; }
            while(left < right && !Char.IsLetterOrDigit(str[right])) { right--; }
            if(Char.ToLower(str[left]) != Char.ToLower(str[right])) { return false; }
            left++; right--;
        }
        return true;
    }
}